tasks/evmRelayMessageWithdrawal.ts

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * ```
+ * yarn hardhat evm-relay-message-withdrawal \
+ * --network [l2_network] --adapter [adapter_address] --spokePool [spoke_pool_address] --value [eth_to_send]
+ * ```
+ * This REQUIRES a spoke pool to be deployed to the specified network AND for the
+ * spoke pool to have the signer as the `crossDomainAdmin`.
+ */
+
+task("evm-relay-message-withdrawal", "Test L1 <-> L2 communication between a deployed L1 adapter and a L2 spoke pool.")
+  .addParam("spokePool", "address of the L2 spoke pool to use.")
+  .addParam(
+    "adapter",
+    "address of the adapter to use. This must correspond to the network on which the L2 spoke pool is deployed"
+  )
+  .addParam("l2Token", "The l2 token address to withdraw from the spoke pool.")
+  .addParam("amountToReturn", "amount of token to withdraw from the spoke pool.")
+  .addOptionalParam(
+    "value",
+    "amount of ETH to send with transaction (which may be needed to call `relayMessage`, such as with zksync). This should only be used in special cases since improper use could nuke funds."
+  )
+  .setAction(async function (taskArguments, hre_) {
+    const hre = hre_ as any;
+    const msgValue = ethers.utils.parseEther(taskArguments.value === undefined ? "0" : value);
+    if (!ethers.utils.isAddress(taskArguments.l2Token))
+      throw new Error(`${taskArguments.l2token} is not a valid evm token address`);
+    if (isNaN(taskArguments.amountToReturn) || taskArguments.amountToReturn < 0)
+      throw new Error(`${taskArguments.amountToReturn} is not a valid amount to send`);
+
+    /**
+     * Setup: Need to obtain all contract addresses involved and instantiate L1/L2 providers and contracts.
+     */
+
+    // Instantiate providers/signers/chainIds
+    const l2ChainId = parseInt(await hre.getChainId());
+    const l1ChainId = parseInt(await hre.companionNetworks.l1.getChainId());
+
+    const l2ProviderUrl = hre.network.config.url;
+
+    const l1Network = l1ChainId === CHAIN_IDs.MAINNET ? "mainnet" : "sepolia";
+    const l1ProviderUrl = hre.config.networks[`${l1Network}`].url;
+
+    const l2Provider = new ethers.providers.JsonRpcProvider(l2ProviderUrl);
+    const l2Signer = ethers.Wallet.fromMnemonic((hre.network.config.accounts as any).mnemonic).connect(l2Provider);
+    const l1Provider = new ethers.providers.JsonRpcProvider(l1ProviderUrl);
+    const l1Signer = ethers.Wallet.fromMnemonic((hre.network.config.accounts as any).mnemonic).connect(l1Provider);
+
+    // Construct the contracts
+    const spokePool = new Contract(taskArguments.spokePool, minimalSpokePoolInterface, l2Signer);
+    const adapter = new Contract(taskArguments.adapter, minimalAdapterInterface, l1Signer);
+
+    console.log("[+] Successfully constructed all contracts. Determining root bundle Id to use.");
+    let rootBundleId = 0;
+    try {
+      while (1) {
+        await spokePool.rootBundles(rootBundleId);
+        rootBundleId++;
+      }
```

Comment:
This might get rate-limited on a free-tier RPC provider. It's probably OK to assume that we'd test this against a very new SpokePool so hopefully it should resolve very quickly and without many calls.